community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreNodeRelationshipCursor.java

```diff
@@ -198,9 +189,11 @@ private long nextChainStart()
     {
         try
         {
-            while ( groupRecord != null )
+            while ( !end )
             {
-                if ( checkType( groupRecord.getType() ) )
+                // We check inUse flag here since we can actually follow pointers in unused records
+                // to guard for and overcome concurrent deletes in the relationship group chain
+                if ( groupRecord.inUse() && checkType( groupRecord.getType() ) )
```

Comment:
Look it at the `RelationshipGroupStore` changes, this appears to be correct after all.